### PR TITLE
Fixes + Features

### PR DESCRIPTION
**Pull request by Skywalker-11**

### CODE REVIEW

File: setup.scripts/mysql/mailwatch-mysql.inc [hunk 1]
```diff
@@ -79,6 +82,11 @@ function setup-database() {
     if [ "$IsUpgrade" == 1 ]; then
         logprint "Not creating sql db because we are running an upgrade"
     elif [ "$mysqlInstalled" == "1" ]; then
+        $(mysql -u"$SqlRoot" -p"$SqlRootPwd" -h "$SqlHost" -e 'SHOW VARIABLES WHERE 0')
+        if [ "$?" != "0" ]; then
```
Comment (Skywalker-11) [inline setup.scripts/mysql/mailwatch-mysql.inc:86]: Check if user can login to mysql server 
(fixes #36)

**Comment by stefaweb** [comment]
Last version block on this:

```
+ echo -e 'Spamassassin not installed. Going to install it'
+ installSpamassassin=y
++ grep -e '^[[:space:]]*bayes_store_module[[:space:]]*Mail::SpamAssassin:BayesStore::MySQL'
```

I had to kill script.

**Comment by Skywalker-11** [comment]
Should already be fixed: https://github.com/mailwatch/mailwatch-install-script/blob/morefixes/setup.scripts/spamassassin/mailwatch-spamassassin.inc#L16

**Comment by stefaweb** [comment]
Forgot to checkout on morefixes

Trying...

**Comment by stefaweb** [comment]
Waiting here. Not clear.
Password of what?

```
Create MailWatch web gui admin
Configure Bayes-Sql
Enter password:
```

EDIT (with root password)
Problem with schema.
```
Configure Bayes-Sql
Enter password:
ERROR 1071 (42000) at line 16: Specified key was too long; max key length is 767 bytes
```

**Comment by Skywalker-11** [comment]
> Waiting here. Not clear.
Password of what?

Have you set a password for the root sql user when asked? It should take that password and don't ask again...
Nevertheless I changed it so that it uses MailWatch credentials.

File: setup.scripts/mysql/mailwatch-mysql.inc [hunk 2]
```diff
@@ -101,4 +110,7 @@ function setup-admin-user() {
         ask "Enter password for the admin: " MWAdminPwd "pw"
         logprint "Create MailWatch web gui admin"
         mysql -u "$SqlUser" -p"$SqlPwd" -h "$SqlHost" "$SqlDb" --execute="REPLACE INTO users SET username = '$MWAdmin', password = MD5('$MWAdminPwd'), fullname = 'Admin', type = 'A';"
```
Comment (stefaweb) [inline setup.scripts/mysql/mailwatch-mysql.inc]: Could you use something better than MD5?
As SHA512-CRYPT

File: setup.scripts/spamassassin/mailwatch-spamassassin.inc [hunk 3]
```diff
@@ -41,7 +45,7 @@ function install-spamassassin() {
 function install-bayesSql() {
     if [ "$installBayesSql" != "n" ]; then
         logprint "Configure Bayes-Sql"
-        mysql -u "$SqlRoot" -p"$SqlRootPwd" -h "$SqlHost" "$SqlDb" < "$InstallFilesFolder/setup.scripts/spamassassin/bayesSql_mysql.sql"
+        mysql -u "$SqlUser" -p"$SqlPass" -h "$SqlHost" "$SqlDb" < "$InstallFilesFolder/setup.scripts/spamassassin/bayes_mysql.sql"
```
Comment (stefaweb) [inline setup.scripts/spamassassin/mailwatch-spamassassin.inc]: Still present:

```
Configure Bayes-Sql
Enter password:
```

Unless when we run setup.sh for the first time.

Can we have:

`Enter password for mailwatch: (Input will not be shown)`

File: setup.scripts/spamassassin/bayes_mysql.sql [hunk 4]
```diff
@@ -24,7 +24,7 @@ CREATE TABLE IF NOT EXISTS `bayes_token` (
 ) ENGINE=InnoDB DEFAULT CHARSET=utf8;
 
 CREATE TABLE IF NOT EXISTS `bayes_vars` (
-`id` int(11) NOT NULL,
+  `id` int(11) NOT NULL AUTO_INCREMENT,
```
Comment (stefaweb) [inline setup.scripts/spamassassin/bayes_mysql.sql]: No longer works.

`ERROR 1075 (42000) at line 26: Incorrect table definition; there can be only one auto column and it must be defined as a key`

**Comment by stefaweb** [comment]
In /etc/MailScanner/spamassassin.conf not good.

- not configured (missing password)
- not at the right place (I'm not sure we can place it at the end)

```
bayes_store_module              Mail::SpamAssassin::BayesStore::MySQL
bayes_sql_dsn                   DBI:mysql:mailscanner:localhost
bayes_sql_username              mailwatch
bayes_sql_password
bayes_sql_override_username     mailwatch
```

File: setup.scripts/mysql/mailwatch-mysql.inc (hunk 2 above)
Comment (Skywalker-11) [reply setup.scripts/mysql/mailwatch-mysql.inc]: Its the same as in the docs
https://docs.mailwatch.org/install/installing.html

Comment (stefaweb) [reply setup.scripts/mysql/mailwatch-mysql.inc]: Its may be something to discuss in mailwatch upstream?
MD5 is old and weak.

**Comment by stefaweb** [comment]
One more.

For `Enter an admin user for the MailWatch web interface:`.

If we forgot to enter a login here, its break the system. Propose a default value (admin) will be better.

File: setup.scripts/mysql/mailwatch-mysql.inc (hunk 2 above)
Comment (Skywalker-11) [reply setup.scripts/mysql/mailwatch-mysql.inc]: yep

File: setup.scripts/mailscanner/00_mw-install-script.conf [hunk 5]
```diff
@@ -0,0 +1,4 @@
+#Default config provided by mailwatch-install-script
+#Do NOT edit this file. It is auto generated.
+#Instead create your own config that arranged after this file in alphabetical order.
+Virus Scanners = clamav
```
Comment (stefaweb) [inline setup.scripts/mailscanner/00_mw-install-script.conf:4]: All these needed to have clamav running:

```
Virus Scanners = clamd
Clamd Socket = /var/run/clamav/clamd.ctl
Clamd Lock File = /var/run/clamav/clamd.pid
Clamd Use Threads = no
```

**Comment by stefaweb** [comment]
Just a cosmetic problem.

See "Clam AV" instead of "ClamAV".
`mailscanner-install.log:Installing Clam AV via apt ...`

**Comment by Skywalker-11** [comment]
You shouldn't realy code if its getting too late 😁 
Now the sql should work

> Just a cosmetic problem.
>
> See "Clam AV" instead of "ClamAV".
mailscanner-install.log:Installing Clam AV via apt ...

Blame MailScanner 😄 and @shawniverson

File: setup.scripts/mailscanner/00_mw-install-script.conf (hunk 5 above)
Comment (Skywalker-11) [reply setup.scripts/mailscanner/00_mw-install-script.conf:4]: If you use `clamav` its not necessary. All you need to do is define it as `Virus Scanners`.
Keep in mind that there is a difference between clamav and clamd:
https://github.com/MailScanner/v5/blob/master/common/etc/MailScanner/virus.scanners.conf

Comment (stefaweb) [reply setup.scripts/mailscanner/00_mw-install-script.conf:4]: Was not working with just "Virus Scanners = clamav"
I restart a complete fresh try. Will see.

Comment (Skywalker-11) [reply setup.scripts/mailscanner/00_mw-install-script.conf:4]: Hmm. I have a server running with default config and its scanning my mails with it.
Have you tried sending an EICAR file to test?

Comment (stefaweb) [reply setup.scripts/mailscanner/00_mw-install-script.conf:4]: OK, it work with last try but its very slow compared to clamd.

Try from GUI with ms_lint.php:

===========================================================================	5.0E-5
Filename Checks: Windows/DOS Executable (1 eicar.com)	0.35477
Other Checks: Found 1 problems	0.05216
Virus and Content Scanning: Starting	0.00054
1.message: Eicar-Test-Signature FOUND	37.38406
./1/eicar.com: Eicar-Test-Signature FOUND	0.00392
Virus Scanning: ClamAV found 2 infections	1.2392
Infected message 1 came from 10.1.1.1	0.00139
Virus Scanning: Found 2 viruses	0.0009
===========================================================================	0.0141
Virus Scanner test reports:	4.0E-5
ClamAV said "eicar.com contains Eicar-Test-Signature"	0.00028
If any of your virus scanners (clamav)

Comment (stefaweb) [reply setup.scripts/mailscanner/00_mw-install-script.conf:4]: Was running during ms_lint.php

www-data 27735 26303  0 11:54 ?        00:00:00 sh -c sudo /usr/sbin/MailScanner --lint 2>&1
postfix  27780 27737  0 11:54 ?        00:00:00 /bin/sh /usr/lib/MailScanner/wrapper/clamav-wrapper /usr -r --infected --stdout .
postfix  27791 27780 99 11:54 ?        00:00:18 /usr/bin/clamscan --tempdir=/var/spool/MailScanner/incoming/clamav-tmp/tmp.y3FUMlcaE0 -r --infected --stdout .

**Comment by stefaweb** [comment]
Could you merge?

I will redo a complete test.

**Pull request merged**